ipatests/test_integration/test_ipahealthcheck.py

```diff
@@ -0,0 +1,507 @@
+        assert data[0]["result"] == "SUCCESS"
+        assert data[0]["kw"]["status"] == "True"
+
+    def test_source_ipahealthcheck_dogtag_ca_dogtagcertsconfigcheck(self):
+        """
+        Testcase checks behaviour of check DogtagCertsConfigCheck in
+        ipahealthcheck.dogtag.ca when tomcat config file is removed
+        """
+        result = self.master.run_command(
+            [
+                "ipa-healthcheck",
+                "--source",
+                "ipahealthcheck.dogtag.ca",
+                "--check",
+                "DogtagCertsConfigCheck",
+                "--output-file",
+                HEALTHCHECK_OUTPUT_FILE,
+            ],
+            raiseonerr=False,
+        )
+        contents = self.master.get_file_contents(
+            HEALTHCHECK_OUTPUT_FILE, encoding="utf-8"
+        )
+        data = json.loads(contents)
+        for check in data:
+            assert check["result"] == "SUCCESS"
+            assert check["kw"]["configfile"] == TOMCAT_CFG
+            assert check["kw"]["key"] in DEFAULT_PKI_CA_CERTS
+        self.master.run_command(["mv", TOMCAT_CFG, TOMCAT_CFG + ".old"])
+        result = self.master.run_command(
+            [
+                "ipa-healthcheck",
+                "--source",
+                "ipahealthcheck.dogtag.ca",
+                "--check",
+                "DogtagCertsConfigCheck",
+                "--output-file",
+                HEALTHCHECK_OUTPUT_FILE,
+            ],
+            raiseonerr=False,
+        )
+        assert result.returncode == 1
+        contents = self.master.get_file_contents(
+            HEALTHCHECK_OUTPUT_FILE, encoding="utf-8"
+        )
+        data = json.loads(contents)
+        assert data[0]["result"] == "CRITICAL"
+        self.master.run_command(["mv", TOMCAT_CFG + ".old", TOMCAT_CFG])
+        self.master.run_command(["ipactl", "restart"])
+
+    def test_source_ipahealthcheck_meta_core_metacheck(self):
+        """
+        Testcase checks behaviour of check MetaCheck in source
+        ipahealthcheck.meta.core when run on IPA master
+        """
+        self.master.run_command(
+            [
+                "ipa-healthcheck",
+                "--source",
+                "ipahealthcheck.meta.core",
+                "--check",
+                "MetaCheck",
+                "--output-file",
+                HEALTHCHECK_OUTPUT_FILE,
+            ],
+            raiseonerr=False,
+        )
+        contents = self.master.get_file_contents(
+            HEALTHCHECK_OUTPUT_FILE, encoding="utf-8"
+        )
+        data = json.loads(contents)
+        assert data[0]["result"] == "SUCCESS"
+        if osinfo.id == "fedora" and osinfo.version_number == (30):
+            assert data[0]["kw"]["ipa_version"] == "4.8.3"
+            assert data[0]["kw"]["ipa_api_version"] == "2.235"
+        else:
+            assert data[0]["kw"]["ipa_version"] == "4.9"
+            assert data[0]["kw"]["ipa_api_version"] == "2.235"
+
```

Comment:
This one test is going to be difficult to maintain because with every new release it will need to be updated as well. It's not a bad test but I'd expect failures. Perhaps instead you could compare that with either the scraped output of ipa --version or by:

result = self.master.run_command(['python3', '-c', 'from ipapython import version; print("%s\t%s" % (version.VERSION, version.API_VERSION))']